From 57c53140db212722945c2e37b1dd94d73fa22c26 Mon Sep 17 00:00:00 2001
From: xiongweimin <xiongweimin@kylinos.cn>
Date: Thu, 16 Jul 2026 11:02:36 +0800
Subject: [PATCH 10405/15333] vhost: reject zero-size IOTLB INVALIDATE

Reject VHOST_IOTLB_INVALIDATE messages with size == 0 to prevent
iova + size - 1 from underflowing to U64_MAX, which would
incorrectly delete the entire IOTLB.

Signed-off-by: xiongweimin <xiongweimin@kylinos.cn>
Signed-off-by: Michael S. Tsirkin <mst@redhat.com>
Message-ID: <20260716030236.124322-1-xiongwm2026@163.com>
---
 drivers/vhost/vhost.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/vhost/vhost.c b/drivers/vhost/vhost.c
index 1be9c0c53bed..a0c1d54019aa 100644
--- a/drivers/vhost/vhost.c
+++ b/drivers/vhost/vhost.c
@@ -1671,6 +1671,10 @@ static int vhost_process_iotlb_msg(struct vhost_dev *dev, u32 asid,
 			ret = -EFAULT;
 			break;
 		}
+		if (!msg->size) {
+			ret = -EINVAL;
+			break;
+		}
 		vhost_vq_meta_reset(dev);
 		vhost_iotlb_del_range(dev->iotlb, msg->iova,
 				      msg->iova + msg->size - 1);
-- 
2.55.0

